pkg/types/aws/validation/platform_test.go

```diff
@@ -128,11 +128,13 @@ func TestValidatePlatform(t *testing.T) {
 				Region: "us-east-1",
 				DefaultMachinePlatform: &aws.MachinePool{
 					EC2RootVolume: aws.EC2RootVolume{
+						Type: "io1",
 						IOPS: -10,
+						Size: 128,
 					},
 				},
 			},
-			expected: `^test-path\.defaultMachinePlatform\.iops: Invalid value: -10: Storage IOPS must be positive$`,
+			expected: `^test-path.*iops: Invalid value: -10: iops must be set for type io1$`,
```

Comment:
Nit: this message looks a bit confusing here. The problem is not that the value must be set, because it is, but that it must be set to a positive value. Is it like this because it defaults to `0` when not set?